From ba0fbc2e9db91cda0986a9721ecf1579dbd58288 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Thu, 9 Jul 2015 08:01:51 +0000
Subject: [PATCH 22/23] net:ThunderX:disable irq migration for vf

Since the irq handler for vf in ThunderX nic driver is bound to
specific CPU, it must set IRQF_NOBALANCING to disable irq
migration.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/net/ethernet/cavium/thunder/nicvf_main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_main.c b/drivers/net/ethernet/cavium/thunder/nicvf_main.c
index 16f83de..f595e57 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_main.c
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_main.c
@@ -761,7 +761,7 @@ static int nicvf_register_interrupts(struct nicvf *nic)
 	for (irq = 0; irq < NICVF_INTR_ID_SQ; irq++) {
 		vector = nic->msix_entries[irq].vector;
 		ret = request_irq(vector, nicvf_intr_handler,
-				  0, nic->irq_name[irq], nic);
+				  IRQF_NOBALANCING, nic->irq_name[irq], nic);
 		if (ret)
 			break;
 		nic->irq_allocated[irq] = true;
-- 
1.7.5.4

